File: back/src/main/java/br/eng/eliseu/presente/model/Tamanho.java
package br.eng.eliseu.presente.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
@Table(name="tamanho")
public class Tamanho {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private ProdutoTipoEnum tipo;

    private String tamanho;

    // Controle de concorrência otimista
    @Version
    private Long version;

}

File: front/src/app/shared/model/tamanho.ts
import { ProdutoTipoEnum } from './enum/produto-tipo.enum';

export class Tamanho {

    id?: number;
    tipo?: ProdutoTipoEnum;
    tamanho?: string;
    // Controle de concorrência otimista (alinhado com o back-end @Version)
    version?: number;

}
